web/tests/upload.test.js

```diff
@@ -224,14 +243,34 @@ test("upload_files", async ({mock_template, override_rewire}) => {
     await upload.upload_files(uppy, config, []);
     assert.ok(!$("#compose-send-button").prop("disabled"));
 
+    // Banners should show even if the "no post permissions" banner
+    // is displayed.
+    const $banner_container = $("#compose_banners");
+    const no_post_permissions_class = `.${CSS.escape(
+        compose_banner.CLASSNAMES.no_post_permissions,
+    )}`;
+    $banner_container.set_find_results(no_post_permissions_class, [{}]);
+
+    const $error_banner = $.create("error_banner");
+    $error_banner.addClass("error");
+
+    const $info_banner = $.create("info_banner");
+    $info_banner.addClass("info");
+
     let banner_shown = false;
+    override_rewire(scroll_util, "get_content_element", () => ({
```

Comment:
I'm not quite sure I understand the need for overriding scroll_util. Could you kindly provide some insight on this matter?

Reply:
Previously, the test cases rely on the fact that the banners have been rendered, through `render_private_stream_warning` and similar functions. Now that we have changed the show logic to reside in `append_compose_banner_to_banner_list`, the render functions will be called regardless of whether the banners are actually shown to the user or not.

From my understanding of the code, the only reliable spot to test if the banner has actually been shown, is on `compose_banner.ts:90`:

```
scroll_util.get_content_element($list_container).append(banner);
```

Hence why I'm overriding `scroll_util` - to check whether the banner is actually being displayed.

Do you maybe have an idea on how this can be done better?

Reply:
Sure thing, got it, thanks!